app/models/round.rb

```diff
@@ -0,0 +1,56 @@
+class Round < ApplicationRecord
+  before_create :set_position
+  before_destroy :update_positions
+
+  validates :round_type, presence: true
+  validate :unique_round_for_player_in_game
+
+  belongs_to :game_player
+  has_one :game, through: :game_player
+
+  enum round_type: {
+    rentz_minus: 'Rentz -',
+    rentz_plus: 'Rentz +',
+    totale_minus: 'Totale -',
+    totale_plus: 'Totale +',
+    king: 'King of hearts',
+    ten: 'Ten of clubs',
+    queens: 'Queens',
+    diamonds: 'Diamonds'
+  }
+
+  def decrement_position
+    self.position -= 1
+    save!
+  end
+
+  private
+
+  def unique_round_for_player_in_game
+    if Round.where(game_player_id:, round_type:)
+            .where.not(id:)
+            .joins(:game_player)
+            .where(game_players: { game_id: game_player.game_id })
+            .any?
```

Comment:
This looks way more complicated than needed.

We either try to validate uniqueness with scope, or
```ruby
Round.where(game_player_id:, round_type:).where.not(id:).any?
```

Reply:
I deleted that method, made managed to validate thru ActiveRecord validations